### DISCUSSION TITLE
Question about Timelords

### DISCUSSION THREAD

**Discussion by narendraj9** [Chia-Network/chia-blockchain#987] [Q&A]
`Proof of time requires a small period of time to pass between blocks. Proof of time is implemented by a Verifiable Delay Function that takes a certain amount of time to compute, but is very fast to verify. The key idea of a VDF is that they require sequential computation, and since having many parallel machines does not yield any benefit, electricity waste is minimized. There will likely be relatively few VDF servers (“Timelords”), as the fastest one will always finish first and it takes only one fast and fair Timelord on the network to complete a block and move the chain forward.`

If the fastest timelord always wins, does it mean it can hijack the blockchain? What is the definition for a fair Timelord in this case? The above is an excerpt from the Chia FAQs.

**Comment by mariano54** [comment]
No, it cannot hijack the blockchain, unless it significantly faster. If the fastest timelord is only slightly faster, it may be able to get a slight benefit from selfish farming (similar to selfish mining in bitcoin). However this requires a massive amount of disk space, on the order of 10% of the total network, and the benefit of rewards they can earn is small.

If the fastest timelord is significantly faster, it can perform a 51% attack more effectively. Basically, if for example your attacker's VDF is twice as fast, then they can perform a 51% attack with only 33% of the total network space. Basically, a 51% attack requires less total space. 
This is one of the reasons why we hope to foster an open research initiative in the future, for making the fastest timelord designs open source and available to all.

**Reply by imbeingfollowed** [reply]
Do I understand correctly that there is no monetary incentive to run timelords? Is it possible to tell how many are currently running?

With few running, I imaging a bit of downtime or a targeted DoS it could be relatively easy be the fastest player?

**Reply by mariano54** [reply]
There are no rewards.
You can tell my crawling the network. 
Farmers have a large large incentive to run a timelord. It's beneficial for many reasons: less latency and thus less chance of being orphaned, good as a backup in bad network conditions, etc. As of now, a well cooled general purpose intel cpu is probably the fastest, but there has been some research into custom hardware for classgroups. I think one of the thing that we should do in the long term is to distribute this hardware to many farmers, to decrease the chances of an attack like you mentioned.

**Reply by turbotas** [reply]
will a network run *at all* if the timelord is too slow?  I have a test network running and the nodes stop working on signage points and require restarting if the timelord is unable to complete the set of 2-wesolowski proofs - they don't finish before the VDF clients get a stop signal.  I'm assuming the timelord is too slow?

**Comment by BrandtH22** [comment]
@narendraj9 , since this ticket was opened a number of years ago without further comments it seems this issue has been resolved. 
We will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions or comments.

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
